Release notes for the interpreter (#229)
crystal-lang/crystal-website: _releases/2022-01-06-1.3.0-released.md

## Interpreter

> The awaited Crystal interpreter has been [merged](https://github.com/crystal-lang/crystal/pull/11159). To use it, you need to compile Crystal with a special flag. The official releases (.deb, .rpm, docker images, etc.) are **not** being compiled with it, since its [missing substantial features](https://github.com/crystal-lang/crystal/issues/11555) for real use cases.

Reviewer: ```suggestion
The awaited Crystal interpreter has been [merged](https://github.com/crystal-lang/crystal/pull/11159). But it's still a preview and [missing substantial features](https://github.com/crystal-lang/crystal/issues/11555) for real use cases. 

Our distribution packages are **not** being compiled with it and we recommend the same for external packages. If you want to try it, you need to build the compiler with interpreter support explicitly (`make crystal interpreter=1`).
```

> The following is an extract of the [post](https://crystal-lang.org/2021/12/29/crystal-i.html) where we introduced this feature.

Reviewer: I would skip that here and instead add `Read more about the interpreter in our blog post: [*Crystal's interpreter – A very special holiday present*](https://crystal-lang.org/2021/12/29/crystal-i.html)` at the end of this section.
```suggestion
```

> Assuming Crystal was compiled with the flag `interpreter=1`, you can invoke the interpreter using two modes: `crystal i` or `crystal i file.cr`. In the first case, a repl will start right away, and in the second case the `file.cr` will be interpreted. Interestingly, if the interpreter finds the `debugger` instruction, it will start an interactive process to debug it. For instance, if we have a file `example.cr` with:

Reviewer: ```suggestion
You can invoke the interpreter using two modes: `crystal i` or `crystal i file.cr`. In the first case, a REPL will start right away, and in the second case the `file.cr` will be interpreted. Interestingly, if the interpreter finds the `debugger` instruction, it will start an interactive process to debug it. For instance, if we have a file `example.cr` with:
```

<div class="code_section">{% highlight crystal %}
a = 1
debugger
p a
{% endhighlight %}</div>

Then calling `crystal i example.cr` stops after the call to `debugger`:
<div class="code_section">{% highlight crystal %}
From: example.cr:3:3 <Program>#example.cr:

    1: a = 1
    2: debugger
 => 3: p a

 pry>
{% endhighlight %}</div>

At this point we can inspect and modify variables, step into function calls, etc. If we enter `a = 2` in the repl and then `next`, we'll see the program printing `2`.

> 

Author: ```suggestion

Read more about the interpreter in our blog post: [*Crystal's interpreter – A very special holiday present*](https://crystal-lang.org/2021/12/29/crystal-i.html)

```
